deeplearning4j-core/src/main/java/org/deeplearning4j/nn/conf/layers/setup/ConvolutionLayerSetup.java

```diff
@@ -147,46 +147,44 @@ private void updateLayerInputs(int i, Layer inputLayer, Layer outputLayer){
                     }
                     break;
                 case "SubsamplingLayer":
-                    if(i < lastLayerNumber){
```

Comment:
@nyghtowl if you go up in the file, you will find that all this code executed under condition `if(i < lastLayerNumber)`, and there is nothing that changes `i` nor `lastLayerNumber` before that. So it just redundant and confusing check.